### PR TITLE
libopae: fix properties sysfs read interface id and socket id

### PR DESCRIPTION
**Pull request by anandaravuri**

### CODE REVIEW

File: libopae/plugins/xfpga/sysfs.c [hunk 1]
```diff
@@ -1045,9 +1047,26 @@ fpga_result get_port_sysfs(fpga_handle handle, char *sysfs_port)
...
+
+	result = opae_glob_path(sysfs_path);
+	if (result) {
+		return result;
+	}
+
+	len = snprintf_s_s(sysfs_port, SYSFS_PATH_MAX, "%s", sysfs_path);
+
+	if (len < 0) {
+		FPGA_ERR("error concatenating strings");
```
[the remark this answers is not in the document]
Comment (anandaravuri) [reply libopae/plugins/xfpga/sysfs.c]: fixed

File: libopae/plugins/xfpga/sysfs.c [hunk 2]
```diff
@@ -1045,9 +1047,26 @@ fpga_result get_port_sysfs(fpga_handle handle, char *sysfs_port)
...
 	}
 
-	snprintf_s_ii(sysfs_port, SYSFS_PATH_MAX,
-		      SYSFS_FPGA_CLASS_PATH SYSFS_AFU_PATH_FMT,
-		      _token->device_instance, _token->subdev_instance);
+	int len  = snprintf_s_s(sysfs_path, SYSFS_PATH_MAX, "%s/../*-port.*",
+		_token->sysfspath);
+
+	if (len < 0) {
+		FPGA_ERR("error concatenating strings");
```
[the remark this answers is not in the document]
Comment (anandaravuri) [reply libopae/plugins/xfpga/sysfs.c]: fixed

File: libopae/plugins/xfpga/properties.c [hunk 3]
```diff
@@ -224,12 +222,14 @@ fpga_result __FPGA_API__ xfpga_fpgaUpdateProperties(fpga_token token,
 	SET_FIELD_VALID(&_iprop, FPGA_PROPERTY_FUNCTION);
 
 	// only set socket id if we have it on sysfs
-	result = sysfs_get_socket_id(_token->device_instance,
-				     _token->subdev_instance,
-				     &_iprop.socket_id);
+	resval = sysfs_parse_attribute64(_token->sysfspath,
+		FPGA_SYSFS_SOCKET_ID, &value);
 	if (0 == result)
```
[the remark this answers is not in the document]
Comment (anandaravuri) [reply libopae/plugins/xfpga/properties.c]: fixed

[the remark this answers is not in the document]
Comment (tswhison) [reply libopae/plugins/xfpga/properties.c]: Shouldn't we return an error here?

[the remark this answers is not in the document]
Comment (anandaravuri) [reply libopae/plugins/xfpga/properties.c]: Fixed

File: libopae/plugins/xfpga/properties.c [hunk 4]
```diff
@@ -224,11 +222,18 @@ fpga_result __FPGA_API__ xfpga_fpgaUpdateProperties(fpga_token token,
 	SET_FIELD_VALID(&_iprop, FPGA_PROPERTY_FUNCTION);
 
 	// only set socket id if we have it on sysfs
-	result = sysfs_get_socket_id(_token->device_instance,
-				     _token->subdev_instance,
-				     &_iprop.socket_id);
-	if (0 == result)
+	result = sysfs_get_fme_path(_token->device_instance,
+	_token->subdev_instance, spath);
```
Comment (r-rojo) [inline libopae/plugins/xfpga/properties.c]: indentation is off

File: libopae/plugins/xfpga/properties.c [hunk 5]
```diff
@@ -224,11 +222,18 @@ fpga_result __FPGA_API__ xfpga_fpgaUpdateProperties(fpga_token token,
...
+	if (FPGA_OK != result)
+			return result;
+
+	resval = sysfs_parse_attribute64(spath,
+		FPGA_SYSFS_SOCKET_ID, &value);
```
Comment (r-rojo) [inline libopae/plugins/xfpga/properties.c]: indentation is off

Comment (anandaravuri) [reply libopae/plugins/xfpga/properties.c]: fixed

File: libopae/plugins/xfpga/properties.c (hunk 4 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/properties.c]: fixed
Pointer to fpga region used in function sysfs_get_fme_path() 
Port & FME objects parent is fpaga region

File: libopae/plugins/xfpga/properties.c [hunk 6]
```diff
@@ -223,9 +223,9 @@ fpga_result __FPGA_API__ xfpga_fpgaUpdateProperties(fpga_token token,
 
 	// only set socket id if we have it on sysfs
 	result = sysfs_get_fme_path(_token->device_instance,
-	_token->subdev_instance, spath);
+			_token->subdev_instance, spath);
```
Comment (r-rojo) [inline libopae/plugins/xfpga/properties.c]: still off

**Pull request merged**